bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/servlet/CameraServlet.java

```diff
@@ -0,0 +1,311 @@
+                    httpService.createDefaultHttpContext());
+        } catch (NamespaceException | ServletException e) {
+            logger.warn("Registering servlet failed:{}", e.getMessage());
+        }
+    }
+
+    @Override
+    protected void doPost(@Nullable HttpServletRequest req, @Nullable HttpServletResponse resp) throws IOException {
+        if (req == null || resp == null) {
+            return;
+        }
+        String pathInfo = req.getPathInfo();
+        if (pathInfo == null) {
+            return;
+        }
+        switch (pathInfo) {
+            case "/ipcamera.jpg":
+                // ffmpeg sends data here for ipcamera.mjpeg streams when camera has no native stream.
+                ServletInputStream snapshotData = req.getInputStream();
+                openStreams.queueFrame(snapshotData.readAllBytes());
+                snapshotData.close();
+                break;
+            case "/snapshot.jpg":
+                snapshotData = req.getInputStream();
+                handler.processSnapshot(snapshotData.readAllBytes());
+                snapshotData.close();
+                break;
+            case "/OnvifEvent":
+                handler.onvifCamera.eventRecieved(req.getReader().toString());
+                break;
+            default:
+                logger.debug("Recieved unknown request \tPOST:{}", pathInfo);
+                break;
+        }
+    }
+
+    @Override
+    protected void doGet(@Nullable HttpServletRequest req, @Nullable HttpServletResponse resp) throws IOException {
+        if (req == null || resp == null) {
+            return;
+        }
+        String pathInfo = req.getPathInfo();
+        if (pathInfo == null) {
+            return;
+        }
+        logger.debug("GET:{}, received from {}", pathInfo, req.getRemoteHost());
+        if (!"DISABLE".equals(handler.getWhiteList())) {
+            String requestIP = "(" + req.getRemoteHost() + ")";
+            if (!handler.getWhiteList().contains(requestIP)) {
+                logger.warn("The request made from {} was not in the whiteList and will be ignored.", requestIP);
+                return;
+            }
+        }
+        switch (pathInfo) {
+            case "/ipcamera.m3u8":
+                Ffmpeg localFfmpeg = handler.ffmpegHLS;
+                if (localFfmpeg == null) {
+                    handler.setupFfmpegFormat(FFmpegFormat.HLS);
+                } else if (!localFfmpeg.getIsAlive()) {
+                    localFfmpeg.startConverting();
+                } else {
+                    localFfmpeg.setKeepAlive(8);
+                    sendFile(resp, pathInfo, "application/x-mpegURL");
+                    return;
+                }
+                // Allow files to be created, or you get old m3u8 from the last time this ran.
+                try {
+                    Thread.sleep(HLS_STARTUP_DELAY_MS);
+                } catch (InterruptedException e) {
+                }
```

Comment:
`doGet()` should be ended as quickly as possible when an `InterruptedException` is thrown.

Reply:
Done